python-cffi/netplan/netdef.py

```diff
@@ -276,17 +316,21 @@ def __next__(self):
 
         # The field 'from' happens to be a reserved keyword in Python
         from_addr = getattr(next_value, 'from')
-        return NetplanRoute(
-            ffi.string(next_value.to).decode('utf-8') if next_value.to else None,
-            ffi.string(next_value.via).decode('utf-8') if next_value.via else None,
-            ffi.string(from_addr).decode('utf-8') if from_addr else None,
-            ffi.string(next_value.type).decode('utf-8') if next_value.type else None,
-            ffi.string(next_value.scope).decode('utf-8') if next_value.scope else None,
-            next_value.family,
-            next_value.table,
-            next_value.metric,
-            next_value.mtubytes,
-            next_value.congestion_window,
-            next_value.advertised_receive_window,
-            next_value.onlink,
-        )
+
+        route = {
+            'to': ffi.string(next_value.to).decode('utf-8') if next_value.to else None,
+            'via': ffi.string(next_value.via).decode('utf-8') if next_value.via else None,
+            'from_addr': ffi.string(from_addr).decode('utf-8') if from_addr else None,
+            'type': ffi.string(next_value.type).decode('utf-8') if next_value.type else None,
+            'scope': ffi.string(next_value.scope).decode('utf-8') if next_value.scope else None,
+            'protocol': None,
+            'table': str(next_value.table) if next_value.table > 0 else 'main',
```

Comment:
As discussed above, we should probably just insert `254` instead of "main" and avoid the string handling in the data-model, this could still be done at the visualisation layer.